core/products/future_for_test.go

```diff
@@ -14,13 +14,20 @@ package products
 
 import (
 	"context"
-	"strconv"
+
+	"code.vegaprotocol.io/vega/libs/num"
+	datapb "code.vegaprotocol.io/vega/protos/vega/data/v1"
 
 	"code.vegaprotocol.io/vega/core/oracles"
 )
 
-func (f *Future) SetSettlementData(ctx context.Context, priceName string, settlementData uint64) {
+func (f *Future) SetSettlementData(ctx context.Context, priceName string, settlementData *num.Numeric, settlementDataType datapb.PropertyKey_Type) {
```

Comment:
Don't really see a reason for passing in the propertykey_type here, The type shouldn't change, so we can can just use `f.oracle.binding.settlementDataPropertyType`, no?

Reply:
Yes, sorry I missed that.
Corrected now.